src/lexer.coffee

```diff
@@ -807,8 +810,9 @@ exports.Lexer = class Lexer
     low = (codePoint - 0x10000) % 0x400 + 0xDC00
     "#{toUnicodeEscape(high)}#{toUnicodeEscape(low)}"
 
-  # Replace \u{...} with \uxxxx[\uxxxx] in strings and regexes
+  # Replace \u{...} with \uxxxx[\uxxxx] in regexes without `u` flag
   replaceUnicodeCodePointEscapes: (str, options) ->
+    shouldReplace = options.flags? and options.flags.indexOf('u') is -1
```

Comment:
@lydell ok yup that reads well, updated

@GeoffreyBooth [requested](https://github.com/jashkenas/coffeescript/pull/4498/files#r112801329) that I refactor `toJS()` into a shared test helper so I opened a pull request #4522 for that (against `master`). Perhaps wait to merge this one until we've gone through the cycle of merging that into `master`, merging `master` into `2`, and re-merging `2` here (or it may not matter if that refactor doesn't cause any merge conflicts here)?